build-with-pip.file

```diff
@@ -1,5 +1,7 @@
 %define my_name %(echo %n | cut -f2 -d-)
-Source: pip://%{my_name}/%{realversion}/source.tar.gz
+#assume that -- will never occur in a pip package name...
+%define pip_name %(echo %my_name | sed 's/--/./g')
+Source: pip://%{pip_name}/%{realversion}/source.tar.gz
```

Comment:
may be instead of putting this logic here, we can have
```
%if "%{?my_name:set}" != "set"
%define my_name %(echo %n | cut -f2 -d-)
%endif
```
and then each package can set it to whatever they want in their spec file.